icu4c/source/test/cintltst/ucptrietest.c

```diff
@@ -1103,6 +1103,47 @@ checkRanges3[]={
     { 0x110000, 6 }
 };
 
+/* set some interesting overlapping ranges */
+/* similar to setRange2 but use only 8 bits value */
```

Comment:
typo: setRange2 --> setRanges2 (plural)

Also, most of the other test cases already exercises 8-bit-value tries, so we should have coverage and should not need to add more test cases. That is, I assume that we need not change this file at all.

Reply:
Actually, I think it would be good to change this test code, but by looking at why it didn't catch this problem, and fixing it.

I think I see the problem: The swapping code is only tested when testTrieRanges() is called with withClone=TRUE. This only happens in TrieTestSet2OverlapWithClone() which uses 16-bit values.

I suggest you just change, say, TrieTestSet3Initial9() to also call testTrieRanges() with withClone=TRUE.